sqlx-sqlite/src/connection/serialized_database.rs

```diff
@@ -0,0 +1,32 @@
+#![allow(unsafe_code)]
+extern crate libsqlite3_sys as ffi;
+
+/// `SerializedDatabase` is a wrapper for a serialized database that is dynamically allocated by calling `sqlite3_serialize`.
+/// This RAII wrapper is necessary to deallocate the memory when it goes out of scope with `sqlite3_free`.
+#[derive(Debug)]
+pub struct SerializedDatabase {
+    data: *mut u8,
+    len: usize,
+}
+
+impl SerializedDatabase {
+    /// Creates a new `SerializedDatabase` with the given data pointer and length.
+    pub fn new(data: *mut u8, len: usize) -> Self {
```

Comment:
This shouldn't be public and should probably be `unsafe`; while it's safe to pass a null pointer to `sqlite3_free()`, passing a pointer that wasn't allocated by SQLite is undefined behavior: https://www.sqlite.org/c3ref/free.html

> The sqlite3_free() routine is a no-op if is called with a NULL pointer. Passing a NULL pointer to sqlite3_free() is harmless. After being freed, memory should neither be read nor written. Even reading previously freed memory might result in a segmentation fault or other severe error. Memory corruption, a segmentation fault, or other severe error might result if sqlite3_free() is called with a non-NULL pointer that was not obtained from sqlite3_malloc() or sqlite3_realloc().